readthedocs/proxito/middleware.py

```diff
@@ -11,21 +11,28 @@
 
 import structlog
 from django.conf import settings
+from django.http import Http404
 from django.shortcuts import redirect, render
 from django.urls import reverse
 from django.utils.deprecation import MiddlewareMixin
 
-from readthedocs.core.unresolver import unresolver
+from readthedocs.core.unresolver import (
+    InvalidCustomDomainError,
+    InvalidExternalDomainError,
+    InvalidSubdomainError,
+    SuspiciousHostnameError,
+    unresolver,
+)
 from readthedocs.core.utils import get_cache_tag
 from readthedocs.projects.models import Domain, Project, ProjectRelationship
 from readthedocs.proxito import constants
 
 log = structlog.get_logger(__name__)  # noqa
 
 
-def map_host_to_project_slug(request):  # pylint: disable=too-many-return-statements
+def map_host_to_project(request):  # pylint: disable=too-many-return-statements
```

Comment:
Should we keep a way to map a host to a project slug still? I suppose there may be situations where we want to know _"What's the project slug for this request (even if the project doesn't exist in our database)?"_

Reply:
If the slug doesn't exist, we should just 404, we shouldn't serve content on `notfound.rtd.io` domains.

Reply:
I agree with that. However, the name of the function and the docstring is confusing. If we are going to use it _only_ in that way, I'd name it `map_host_to_project_or_404` and expand the docstring to make it clear that it raises an exception.